File: algos_assignment/3rd_grade/이서진_metamong/2nd_week1/18258.py
import sys
from collections import deque

queue=deque()
def push(x):
    queue.append(x)
def pop():
    if(not queue):
        return -1
    else :
        return queue.popleft()
def size():
    return len(queue)
def empty():
    if(not queue):
        return 1
    else :
        return 0
def front():
    if(not queue): return -1
    else:
        return queue[0]
def back():
    if(not queue): return -1
    else:
        return queue[-1]

n=int(sys.stdin.readline().rstrip())
for i in range(n):
    word=sys.stdin.readline().split()
    order=word[0]
    if order=="push":
        push(word[1])
    elif order=="pop":
        print(pop())
    elif order=="size":
        print(size())
    elif order=="empty":
        print(empty())
    elif order=="front":
        print(front())
    elif order=="back":
        print(back())
    

File: algos_assignment/2nd_grade/장아현_becherryu/week1/18258.cpp
#include <iostream>
#include <queue>

using namespace std;

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(0); cout.tie(0);
	int n, a;
	cin >> n;

    queue<int> q;
	string str;

	for(int i = 0; i < n; i++) {
		cin >> str;
		if (str == "push") {
			cin >> a;
			q.push(a);
		}
		else if (str == "pop") {
			if (q.empty())
				cout << "-1\n";
			else {
                cout << q.front() << "\n";
                q.pop();
            }
		}
		else if (str == "size")
			cout << q.size() << "\n";

		else if (str == "empty") {
			if (q.empty())
				cout << "1\n";
			else
				cout << "0\n";
		}
		else if (str == "front") {
			if (q.empty())
				cout << "-1\n";
			else
				cout << q.front() << "\n";
		}
		else //back
			if (q.empty())
				cout << "-1\n";
			else
				cout << q.back() << "\n";
	}
    return 0;
}
